Kata 7, "Fibonacci=)=)=)=)=)".

udodikdmitro opened the pull request:
To be honest, I haven't looked on the Internet.
I need answers to my some questions about
solutions from other peoples.

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
To be honest, I haven't looked on the Internet.
I need answers to my some questions about
solutions from other peoples.

ezseeton commented on src/katatraining/fibonacci/Fibonacci.java:
```diff
@@ -0,0 +1,13 @@
+
+public class Fibonacci {
+    public static long fib(int n) {
+        int[] x = new int[n + 1];
+        x[0] = 0;
+        x[1] = 1;
```
You set x[0] = 0. Does it mean that someone could use your method to find element with index zero? If it's so then you will get an exception. 

Just write another one test in FibonacciTest class for that case and you'll see where is the problem.

Fibonacci.fib(0)

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
Add 1 line into test.
Exception is solved, but I don't understand this
solution fully...

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
Little error corrected.

udodikdmitro commented:
PS: 4.00 o'clock - this was evening 15.08 or morning 16.08???))

ezseeton commented on src/katatraining/fibonacci/Fibonacci.java:
```diff
@@ -0,0 +1,13 @@
+
+public class Fibonacci {
+    public static long fib(int n) {
+        int[] x = new int[n + 2];
+        x[0] = 0;
+        x[1] = 1;
+        for (int i = 2; i <= n; i++) {
+            x[i] = x[i - 1] + x[i - 2];
+        }
+        return x[n];
```
Yes, sure, your fix is simple. But I think we don't need to create an array at all before your aswer equals incomming parameter. Let's see:
n = 0, x[n] = 0
n = 1, x[n] = 1

so, maybe, if n <2, you should return n? It would safe our memory and running time.

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
Solution of this kata without creating array.

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
Code was shortened.

Commit:
Kata 7, "Fibonacci=)=)=)=)=)".
Optimization.
